From ccdd226ca0f93e7c4713a7c2a753dbed29d2dc40 Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Fri, 16 Apr 2021 16:26:05 +0800
Subject: [PATCH] Allow all users can receive ACTION_USER_UNLOCKED

We will start user dynamically in concurrent multi user
mode, concurrent user service need to monitor user status
changed, but it always run as system uid, so allow all
users can receive ACTION_USER_UNLOCKED.

Change-Id: Ica807887a5f5a9177b83bc99a4821641a8312f2b
Tracked-On: ACP-1373
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 services/core/java/com/android/server/am/UserController.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/UserController.java b/services/core/java/com/android/server/am/UserController.java
index 2d8d2c32a4c..afb3ca25b81 100644
--- a/services/core/java/com/android/server/am/UserController.java
+++ b/services/core/java/com/android/server/am/UserController.java
@@ -623,7 +623,7 @@ class UserController implements Handler.Callback {
                 Intent.FLAG_RECEIVER_REGISTERED_ONLY | Intent.FLAG_RECEIVER_FOREGROUND);
         mInjector.broadcastIntent(unlockedIntent, null, null, 0, null,
                 null, null, AppOpsManager.OP_NONE, null, false, false, MY_PID, SYSTEM_UID,
-                Binder.getCallingUid(), Binder.getCallingPid(), userId);
+                Binder.getCallingUid(), Binder.getCallingPid(), UserHandle.USER_ALL);
 
         if (getUserInfo(userId).isManagedProfile()) {
             UserInfo parent = mInjector.getUserManager().getProfileParent(userId);
-- 
2.17.1

